plugins/by-name/rest/default.nix

```diff
@@ -399,4 +402,31 @@ helpers.neovim-plugin.mkNeovimPlugin {
       ]
     ];
   };
+
+  extraOptions = {
+    curlPackage = lib.mkPackageOption pkgs "curl" {
+      nullable = true;
+    };
+
+    enableHttpFiletypeAssociation = lib.mkEnableOption "filetype association of http files";
```

Comment:
Is the plugin useful without this filetype? If yes, this may be correct. If 90% of users will want it, maybe we make an exception to our usual "everything must by explicit" policy and set a sane-default here?

Reply:
Oh yeah, this defaults to false doesn't it...

Reply:
We can either `// {default = true;}` or use `mkOption`